src/AWS.Deploy.Orchestration/LocalUserSettings/LocalUserSettingsEngine.cs

```diff
@@ -0,0 +1,208 @@
+            }
+            catch (Exception ex)
+            {
+                throw new FailedToUpdateLocalUserSettingsFileException($"Failed to update the local user settings file " +
+                    $"to include the last deployed to stack '{stackName}'.", ex);
+            }
+        }
+
+        /// <summary>
+        /// This method updates the local user settings json file by deleting the stack that was most recently used.
+        /// </summary>
+        public async Task DeleteLastDeployedStack(string stackName, string projectName, string? awsAccountId, string? awsRegion)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(projectName))
+                    throw new FailedToUpdateLocalUserSettingsFileException("The Project Name is not defined.");
+                if (string.IsNullOrEmpty(awsAccountId) || string.IsNullOrEmpty(awsRegion))
+                    throw new FailedToUpdateLocalUserSettingsFileException("The AWS Account Id or Region is not defined.");
+
+                var localUserSettings = await GetLocalUserSettings();
+                var lastDeployedStack = localUserSettings?.LastDeployedStacks?
+                    .FirstOrDefault(x => x.Exists(awsAccountId, awsRegion, projectName));
+
+                if (localUserSettings == null || lastDeployedStack == null)
+                    return;
+
+                lastDeployedStack.Stacks.Remove(stackName);
+
+                await WriteLocalUserSettingsFile(localUserSettings);
+            }
+            catch (Exception ex)
+            {
+                throw new FailedToUpdateLocalUserSettingsFileException($"Failed to update the local user settings file " +
+                    $"to delete the stack '{stackName}'.", ex);
+            }
+        }
+
+        /// <summary>
+        /// This method updates the local user settings json file by deleting orphan stacks.
+        /// </summary>
+        public async Task CleanOrphanStacks(List<string> deployedStacks, string projectName, string? awsAccountId, string? awsRegion)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(projectName))
+                    throw new FailedToUpdateLocalUserSettingsFileException("The Project Name is not defined.");
+                if (string.IsNullOrEmpty(awsAccountId) || string.IsNullOrEmpty(awsRegion))
+                    throw new FailedToUpdateLocalUserSettingsFileException("The AWS Account Id or Region is not defined.");
+
+                var localUserSettings = await GetLocalUserSettings();
+                var localStacks = localUserSettings?.LastDeployedStacks?
+                    .FirstOrDefault(x => x.Exists(awsAccountId, awsRegion, projectName));
+
+                if (localUserSettings == null || localStacks == null)
+                    return;
+
+                var validStacks = deployedStacks.Intersect(localStacks.Stacks);
+
+                localStacks.Stacks = validStacks.ToList();
```

Comment:
Where is `validStacks` used? Possibly both statements can be removed.
```suggestion
```

Reply:
validStacks is being set to  localStacks.Stacks
So we are basically getting the intersection, setting it in the model and then writing the file back.

Reply:
Makes sense, it was hard to follow in one go.